server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeService.java

```diff
@@ -238,4 +239,17 @@ public static boolean isMigratingToRemoteStore(ClusterSettings clusterSettings)
 
         return (isMixedMode && isRemoteStoreMigrationDirection);
     }
+
+    /**
+     * To check if the cluster is undergoing remote store migration using metadata settings
+     * @return
+     * <code>true</code> For <code>REMOTE_STORE</code> migration direction and <code>MIXED</code> compatibility mode,
+     * <code>false</code> otherwise
+     */
+    public static boolean isMigratingToRemoteStore(Settings settings) {
```

Comment:
this doesn't look correct. `Settings` is very generic and can be any settings (cluster/index). Can we take `Metadata` as an input ?